src/sinks/prometheus/exporter.rs

```diff
@@ -920,11 +956,17 @@ mod tests {
 
         assert!(result.status().is_success());
 
+        if encoding.is_some() {
+            assert!(result
+                .headers()
+                .contains_key(http::header::CONTENT_ENCODING));
+        }
+
         let body = result.into_body();
         let bytes = hyper::body::to_bytes(body)
             .await
             .expect("Reading body failed");
-        let result = String::from_utf8(bytes.to_vec()).unwrap();
+        let result = unsafe { String::from_utf8_unchecked(bytes.to_vec()) };
```

Comment:
A compressed body won't be valid UTF-8 but we can use this unchecked version to still get a String - the only caveat is the `unsafe` block. Doesn't seem like a big deal IMO since its test code and the other uses are asserting the body is as expected - open to other ideas :)